specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11-preview/searchservice.json

```diff
@@ -3956,6 +4152,329 @@
+			},
+			"skills": {
+				"type": "array",
+				"items": {
+					"$ref": "#/definitions/Skill"
+				},
+				"description": "A list of skills in the skillset."
+			},
+			"@odata.etag": {
+				"x-ms-client-name": "ETag",
+				"type": "string",
+				"description": "The ETag of the skillset."
+			}
+		},
+		"required": ["name",
+		"description",
+		"skills"],
+		"description": "A list of Cognitive Search skills."
+	},
+	"Skill": {
+		"discriminator": "@odata.type",
+		"properties": {
+			"@odata.type": {
+				"type": "string"
+			},
+			"description": {
+				"type": "string",
+				"description": "The description of the skill."
+			},
+			"context": {
+				"type": "string",
+				"description": "The context of the skill."
+			},
+			"inputs": {
+				"type": "array",
+				"items": {
+					"$ref": "#/definitions/InputFieldMappingEntry"
+				},
+				"description": "The inputs of the skill."
+			},
+			"outputs": {
+				"type": "array",
+				"items": {
+					"$ref": "#/definitions/OutputFieldMappingEntry"
+				},
+				"description": "The outputs of the skill."
+			}
+		},
+		"required": ["@odata.type",
+		"description",
+		"context",
+		"inputs",
+		"outputs"],
+		"description": "Abstract base class for skills."
+	},
+	"InputFieldMappingEntry": {
+		"properties": {
+			"name": {
+				"type": "string",
+				"description": "The name of the input."
+			},
+			"source": {
+				"type": "string",
+				"description": "The source of the input."
+			}
+		},
+		"required": ["name",
+		"source"],
+		"description": "Input field mapping for a skill."
+	},
+	"OutputFieldMappingEntry": {
+		"properties": {
+			"name": {
+				"type": "string",
+				"description": "The name of the output."
+			},
+			"targetName": {
+				"type": "string",
+				"description": "The target name of the output."
+			}
+		},
+		"required": ["name",
+		"targetName"],
+		"description": "output field mapping for a skill."
+	},
+	"KeyPhraseExtractionSkill": {
+		"x-ms-discriminator-value": "#Microsoft.Skills.Text.KeyPhraseExtractionSkill",
+		"allOf": [{
+			"$ref": "#/definitions/Skill"
+		}],
+		"properties": {
+			"defaultLanguageCode": {
+				"type": "string",
+				"default": "en",
+				"description": "A value indicating which language code. Default is en."
+			},
+			"maxKeyPhraseCount": {
+				"type": "integer",
+				"format": "int32",
+				"x-nullable": true
+			}
+		},
+		"description": "Text analytics key phrase extraction."
+	},
+	"OcrSkill": {
+		"x-ms-discriminator-value": "#Microsoft.Skills.Vision.OcrSkill",
+		"allOf": [{
+			"$ref": "#/definitions/Skill"
+		}],
+		"properties": {
+			"textExtractionAlgorithm": {
+				"type": "string",
+				"default": "printed",
+				"description": "A value indicating which algorithm to use. Default is printed."
+			},
+			"defaultLanguageCode": {
+				"type": "string",
+				"default": "en",
+				"description": "A value indicating which language code. Default is en."
+			},
+			"detectOrientation": {
```

Comment:
This property name works fine for a REST API, but on the client side we could do better. For example, the .NET naming guidelines suggest starting Boolean property names with a prefix like "Is", "Has", "Should", etc. If you want to use a better name in the generated SDK code, you can use `x-ms-client-name`.

Reply:
Thanks a lot for the tips. "x-ms-client-name": "ShouldDetectOrientation" is added.